lib/rules/jsx-newline.js

```diff
@@ -35,6 +48,27 @@ module.exports = {
             if (element.type === 'JSXElement' || element.type === 'JSXExpressionContainer') {
               const firstAdjacentSibling = elements[index + 1];
               const secondAdjacentSibling = elements[index + 2];
+              if (configuration.removeNewLines) {
+                if (
+                  firstAdjacentSibling
+                && secondAdjacentSibling
+                && (firstAdjacentSibling.type === 'Literal' || firstAdjacentSibling.type === 'JSXText')
+                // Check adjacent sibling has new lines
+                && /\n\s*\n/.test(firstAdjacentSibling.value)
+                ) {
+                  context.report({
+                    node: secondAdjacentSibling,
+                    messageId: 'noNewLine',
+                    fix(fixer) {
+                      return fixer.replaceText(
+                        firstAdjacentSibling,
+                        sourceCode.getText(firstAdjacentSibling)
+                          .replace(/\n/g, '')
```

Comment:
@ljharb This regex looks a bit sketchy. I guess it should just be like one `\n` rather than many

Reply:
line endings could also be `\r\n` or whichever, i suppose

Reply:
Yeah